utils.js

```diff
@@ -0,0 +1,66 @@
+function shuffle(array) {
+  array = array.slice(); // Shallow clone array
+  let currentIndex = array.length, temporaryValue, randomIndex;
+
+  // While there remain elements to shuffle...
+  while (currentIndex !== 0) {
+
+    // Pick a remaining element...
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+
+    // And swap it with the current element.
+    temporaryValue = array[currentIndex];
+    array[currentIndex] = array[randomIndex];
+    array[randomIndex] = temporaryValue;
+  }
+
+  return array;
+}
+
+module.exports.shuffle = shuffle;
+
+function removeRanges(array, ranges) {
+  array = array.slice(); // Shallow clone array
+
+  ranges.sort(
```

Comment:
🌹

Reply:
At any other point in life, it may be worth breaking this out into another function so it can be better tested. Not needed right now, but I'm having random thoughts.